### PR TITLE
Boost 1.61.0 support and set as default

### PR DESCRIPTION
**Pull request by dan-42**
Adding Boost V1.61.0 released on May 13th, 2016
And sets it as default Version.

### CODE REVIEW

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.430
* https://travis-ci.org/ingenue/hunter/builds/130545431

File: cmake/projects/Boost/hunter.cmake [hunk 1]
```diff
@@ -17,6 +17,18 @@ hunter_add_version(
     PACKAGE_NAME
     Boost
     VERSION
+    "1.61.0"
+    URL
+    "https://github.com/hunter-packages/boost/archive/v1.61.0.tar.gz"
```
Comment (ruslo) [inline cmake/projects/Boost/hunter.cmake]: Well there is no such version :) All tests fail.

**Comment by dan-42** [comment]
Ups...  that was  typo.  Did't pay attention when copyed from my test-branch.
It does makes sense though stripping the offical release from docs and tests.

Do you have a documentation on how I can provide you with striped version of boost?

**Comment by ruslo** [comment]
> Do you have a documentation on how I can provide you with striped version of boost?

Usually I do it myself. But since optimal-size Boost version can be used with binaries cache feature there is no difference between using stripped or official sources.

**Comment by ruslo** [comment]
New testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.432
* https://travis-ci.org/ingenue/hunter/builds/130565724

**Comment by ruslo** [comment]
> New testing

iOS testing failed. This is because code compilation triggered on header-only install (header-only should just copy files). The reason is that new `metaparse` component was introduced in `1.61`. Please add `metaparse` to:
* https://github.com/ruslo/hunter/blob/e4b0df727021fec5da45a21a617903d4145d83ab/cmake/projects/Boost/schemes/url_sha1_boost.cmake.in#L43
* https://github.com/ruslo/hunter/blob/e4b0df727021fec5da45a21a617903d4145d83ab/cmake/projects/Boost/schemes/url_sha1_boost_ios_library.cmake.in#L49
* https://github.com/ruslo/hunter/blob/e4b0df727021fec5da45a21a617903d4145d83ab/cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in#L48

**Comment by dan-42** [comment]
I thought for header only there is nothing to do. I missed that to.

all new 4 libs are header only, but e.g. hanna requires gcc >= 6.0
 * hana  
 * dll
 * metaparse
 * compute

@ruslo I hope the tests run now, otherwiese I'll first try to setup the test by my own. Before bothering you.

File: cmake/projects/Boost/schemes/url_sha1_boost.cmake.in [hunk 2]
```diff
@@ -42,19 +42,23 @@ endmacro()
 
 boost_component(atomic 1.53.0)
 boost_component(chrono 1.47.0)
+boost_component(compute 1.61.0)
```
Comment (ruslo) [inline cmake/projects/Boost/schemes/url_sha1_boost.cmake.in]: We don't need to list header-only components here. The goal is to have "not building" everywhere when Boost header-only part is building. For now it looks like this (see [log](https://travis-ci.org/ingenue/hunter/jobs/130565757#L984)):
```
Component configuration:

    - atomic                   : not building
    - chrono                   : not building
    - container                : not building
    - context                  : not building
    - coroutine                : not building
    - coroutine2               : not building
    - date_time                : not building
    - exception                : not building
    - filesystem               : not building
    - graph                    : not building
    - graph_parallel           : not building
    - iostreams                : not building
    - locale                   : not building
    - log                      : not building
    - math                     : not building
    - metaparse                : building
    - mpi                      : not building
    - program_options          : not building
    - python                   : not building
    - random                   : not building
    - regex                    : not building
    - serialization            : not building
    - signals                  : not building
    - system                   : not building
    - test                     : not building
    - thread                   : not building
    - timer                    : not building
    - type_erasure             : not building
    - wave                     : not building
```

I.e. you only need to add `boost_component(metaparse 1.61.0)`

**Comment by dan-42** [comment]
@ruslo updated once again. Thank you

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.436
* https://travis-ci.org/ingenue/hunter/builds/130682666

**Comment by ruslo** [comment]
> Testing

Looks good. However some windows tests failed because of network problems and AppVeyor is not capable to restart one toolchain independently (only all of them) so I've created separate branch:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.445

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.448
* https://travis-ci.org/ingenue/hunter/builds/131086462

**Comment by dan-42** [comment]
@ruslo awesome thank you.

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.14.22

**Pull request merged**